File: Unilink/Flutter/lib/repositories/Impl/TopicRepository.dart
import 'package:unilink_flutter_app/models/topic_model.dart';
import 'package:unilink_flutter_app/repositories/ITopicRepository.dart';
import 'package:unilink_flutter_app/service/topic_service.dart';

class TopicRepository implements ITopicRepository {
  TopicService service = TopicService();
  @override
  Future<List<Topic>> getAll() {
    return service.getAll();
  }

  Future<List<Topic>> getTopicByParty(String partyId, String pageSize,
      String curPage, String sortBy, String sortType) async {
    try {
      return service.getTopicByParty(
          partyId, pageSize, curPage, sortBy, sortType);
    } catch (e) {
      throw Exception(e.toString());
    }
  }

  @override
  Future createTopic(String title, String description, String partyId) async {
    try {
      return await service.createTopic(title, description, partyId);
    } catch (e) {
      throw Exception(e.toString());
    }
  }

  @override
  Future updateTopic(String title, String description, String topicId) async {
    try {
      return await service.updateTopic(title, description, topicId);
    } catch (e) {
      throw Exception(e.toString());
    }
  }

  @override
  Future deleteTopic(String topicId) async {
    try {
      return await service.deleteTopic(topicId);
    } catch (e) {
      throw Exception(e.toString());
    }
  }
}

File: Unilink/DotNet/Unilink.Data/Repository/Impl/TopicRepository.cs
using EntityFrameworkPaginateCore;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Unilink.Data.Config;
using Unilink.Data.Dto.Pagination;
using Unilink.Data.Entity;

namespace Unilink.Data.Repository.Impl
{
    public class TopicRepository : GenericRepository<Topic>, ITopicRepository
    {
        private readonly ApplicationDbContext _DBContext;
        public TopicRepository(ApplicationDbContext context) : base(context)
        {
            this._DBContext = context;
        }

        public async Task<Page<Topic>> searchPagination(PaginationRequestDTO searchDTO)
        {
            try
            {
                var filters = new Filters<Topic>();
                if (!string.IsNullOrEmpty(searchDTO.searchText))
                {
                    Guid guidOutput;
                    bool isValid = Guid.TryParse(searchDTO.searchText, out guidOutput);
                    filters.Add(isValid, x => x.PartyId.ToString().Equals(searchDTO.searchText));
                }
                //Filter
                
                filters.Add(true, x => x.Status == true);

                //Sort
                var sorts = new Sorts<Topic>();
                if (searchDTO.sortType.Equals("dsc"))
                {
                    sorts.Add(searchDTO.sortBy.Equals("Title"), x => x.Title, true);
                    sorts.Add(searchDTO.sortBy.Equals("Id"), x => x.Id, true);
                    sorts.Add(searchDTO.sortBy.Equals("CreateDate"), x => x.CreateDate, true);
                    sorts.Add(searchDTO.sortBy.Equals("PartyId"), x => x.PartyId, true);
                }
                else
                {
                    sorts.Add(searchDTO.sortBy.Equals("Title"), x => x.Title, false);
                    sorts.Add(searchDTO.sortBy.Equals("Id"), x => x.Id, false);
                    sorts.Add(searchDTO.sortBy.Equals("CreateDate"), x => x.CreateDate, false);
                    sorts.Add(searchDTO.sortBy.Equals("PartyId"), x => x.PartyId, false);
                }

                return await _DBContext.Topics.Select(e => e).PaginateAsync(searchDTO.curPage, searchDTO.pageSize, sorts, filters);
            }
            catch (Exception ex)
            {
                return null;
            }
        }
    }
}
